integration_tests/src/main/python/date_time_test.py

```diff
@@ -215,6 +215,21 @@ def test_to_unix_timestamp(data_gen, ansi_enabled):
         {'spark.sql.ansi.enabled': ansi_enabled})
 
 
+@pytest.mark.parametrize('time_zone', ["UTC", "UTC+0", "UTC-0", "GMT", "GMT+0", "GMT-0"], ids=idfn)
+@pytest.mark.parametrize('data_gen', [timestamp_gen], ids=idfn)
+def test_from_utc_timestamp(data_gen, time_zone):
+    assert_gpu_and_cpu_are_equal_collect(
+        lambda spark : unary_op_df(spark, data_gen).selectExpr("from_utc_timestamp(a, '{}')".format(time_zone)))
+
+@allow_non_gpu('ProjectExec, FromUTCTimestamp')
+@pytest.mark.parametrize('time_zone', ["PST", "MST", "EST", "VST", "NST", "AST"], ids=idfn)
+@pytest.mark.parametrize('data_gen', [timestamp_gen], ids=idfn)
+def test_from_utc_timestamp_fallback(data_gen, time_zone):
+    assert_gpu_fallback_collect(
+        lambda spark : unary_op_df(spark, data_gen).selectExpr("from_utc_timestamp(a, '{}')".format(time_zone)),
```

Comment:
non-blocking nit: this could be more type-safe using `select` API
```suggestion
        lambda spark : unary_op_df(spark, data_gen).select(f.from_utc_timestamp(a, time_zone)),
```